[AArch64] Avoid using NEON FCVTXN in Streaming-SVE mode.

sdesmalen-arm opened the pull request:
We can still lower these operations using (streaming-compatible) SVE
instructions when compiling for SME or SVE2.

sdesmalen-arm committed:
[AArch64] Avoid using NEON FCVTXN in Streaming-SVE mode.

We can still lower these operations using (streaming-compatible) SVE
instructions when compiling for SME or SVE2.

aemerson reviewed (approved):
Seems reasonable to me.

paulwalker-arm commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -19507,7 +19507,27 @@ static SDValue performBuildVectorCombine(SDNode *N,
+  const auto &Subtarget = DAG.getSubtarget<AArch64Subtarget>();
+  bool CanUseFCVTXN = Subtarget.isNeonAvailable() ||
+                      (Subtarget.useSVEForFixedLengthVectors() &&
+                       (Subtarget.hasSVE2() || Subtarget.hasSME()));
```
This isn't sufficient to say we're in streaming mode is it?  Don't we need something like `|| (Subtarget.hasSME() && Subtarget.isStreaming())`

paulwalker-arm reviewed (commented):
Out of interest how bad is the code if you just disable the combine when NEON is not available.  I ask because given the IR the current code generation looks pretty poor so the combine doesn't seem all that useful in the long run.

sdesmalen-arm committed:
Don't emulate FCVTXN with SVE

@paulwalker-arm requested to simplify the code for now since the SVE code is already
not great and this is something we'll want to change separately.
